### ISSUE TITLE
Cannot send email with development version (1.8-develop)

### ISSUE THREAD

**Issue by HelmutL** [fisharebest/webtrees#1259]
I cannot send Emails (using PHP Mail as SMTP is forbidden by the server). There is an error:

2017-06-20 08:04:35	error	Mail: /users/hlein/www/webtrees-develop/vendor/swiftmailer/swiftmailer/lib/preferences.php:17 is_writable() [<a href='http://de.php.net/function.is-writable'>function.is-writable</a>]: open_basedir restriction in effect. File(/tmp) is not within the allowed path(s): (/users/hlein/www:/users/_temp/hlein)

NB:
the directory data/tmp has been created automatically with access rights 755

Note:
my production is running webtrees 1.7.7 (without SwiftMail) and is ok.

**Comment by fisharebest** [comment]
This is a server configuration issue.

Your server is configured to use the folder `/tmp` for temporary files.  You can check this in your PHPinfo page.

Your server is also configured to trigger an error when a script tries to access this folder!

You *might* be able to suppress this error by editing `includes/session.php` and changing the definition of `WT_DEBUG` to `false`.

**Comment by HelmutL** [comment]
Maybe I could avoid the rror message. But I cannot change the  server configuration. So, the mail is not sent.

**Comment by fisharebest** [comment]
I do not have access to a server with the `open_basedir_restriction`.

> editing `includes/session.php` and changing

I am asking you to test this for me.

**Comment by HelmutL** [comment]
if I am setting WT_DEBUG to false, no errors are written into the log. But still, the mail is not sent (although webtrees thinks is it ok - "The message was successfully sent to xxx" - the samebehaviour as before)

**Comment by HelmutL** [comment]
I copied and adapted an example from selfhtml and uploaded it to the server ("my" root directory). And it sends mails. I attach the script.
[testmail.zip](https://github.com/fisharebest/webtrees/files/1097261/testmail.zip)

**Comment by melizaa** [comment]
I started to get the error after my server made some changes.

I removed locally the block of code at line 17 in webtrees/vendor/swiftmailer/swiftmailer/lib/preferences.php. Since then mails are sent. I have not seen any issues with the memory.

Meliza

**Comment by HelmutL** [comment]
strange. This does not help here (removed the if block lines 17-19)

**Comment by fisharebest** [comment]
> This does not help here (removed the if block lines 17-19)

You deleted line 17-19.

You still get exactly the same error on line 17 ???

> strange

Indeed.

I would say it is not possible.  Are you 100% certain?

**Comment by HelmutL** [comment]
I am rather sure.
webtrees says the message is sent successfully, but the server does  not send it out.  Since some time, I do not see an error message in the webtrees log. It seems it  first disappeared when  changing the session.php. But it did not reappear afrter a change back.
To be sure, I deleted (dropped) the database anddeleted the webtrees-develop subtree. Then, I recreated webtrees-develop and the database and loaded it. I modified the swiftmailer preferences.

Still, the behaviour is:webtrees says the message is sent successfully, no error message in the webtrees log, the message is not sent. Again, the restmail script from above still works.

Swiftmailer preferenses:
<?php

/****************************************************************************/
/*                                                                          */
/* YOU MAY WISH TO MODIFY OR REMOVE THE FOLLOWING LINES WHICH SET DEFAULTS  */
/*                                                                          */
/****************************************************************************/

$preferences = Swift_Preferences::getInstance();

// Sets the default charset so that setCharset() is not needed elsewhere
$preferences->setCharset('utf-8');

// Without these lines the default caching mechanism is "array" but this uses a lot of memory.
// If possible, use a disk cache to enable attaching large attachments etc.
// You can override the default temporary directory by setting the TMPDIR environment variable.
//if (@is_writable($tmpDir = sys_get_temp_dir())) {
//)    $preferences->setTempDir($tmpDir)->setCacheType('disk');
//}

// this should only be done when Swiftmailer won't use the native QP content encoder
// see mime_deps.php
if (PHP_VERSION_ID < 50407) {
    $preferences->setQPDotEscape(false);
}

**Comment by fisharebest** [comment]
You have selected "PHP Mail".

This means that you will use the mail transport that is built into your version of PHP.
It could use SMTP.  It could use sendmail.  It could use something else.

The PHP mail function provides no feedback.  If it cannot send the mail, it still returns "success".

If PHP mail cannot send mail, then the only person who can help is your server administrator.

Also, the PHP mail function has security issues, and most mail libraries are dropping support for it.

You *really* should use SMTP.

You have a webhost that has mis-configured your server (tmp folder is inaccessible), and will not allow you to use SMTP.  Perhaps find another webhost?

**Comment by fisharebest** [comment]
Create a folder `data/tmp`
Makde sure it is writable (e.g. owned by web process or mode = 777);

Add this line to the end of `includes\session.php`

```putenv('TMPDIR=' . WT_DATA_DIR . 'tmp');```

**Comment by HelmutL** [comment]
Applied the above, but nothing changed.

Now it starts to be **really** strange.
I inserted a line into app/Mail.php to see the value of the transport (write it to the error log):
`				->addPart(Filter::unescapeHtml($message), 'text/plain');

      Log::addErrorLog('Mail: transport: ' . Site::getPreference('SMTP_ACTIVE'));

			Swift_Mailer::newInstance(self::transport())->send($mail);
		} catch (Exception $ex) {
			Log::addErrorLog('Mail: ' . $ex->getMessage());
`
**But nothing is writtten to the log !!** Does it mean that the send  function in Mail.php is not executed ?

**Comment by HelmutL** [comment]
I have found something:
I inserted a line in message.php before line 253 to see the contact method.
It showd it was 'messaging' which was not tested for in the if block -> Mail::send is not called, which explains trhe strange behaviour.

I changed line 253 to include also 'messaging', so Mail::send is called.

Result:
Mail not sent ("The  message was not sent") and error logged in the log table (I cannot see the error using Website log - search error because there is no userid (= NULL) which cannot be selected in the form, but I can see it using phpmyadmin).
2 cases:
1)- Mail transport internal (PHP) results in
Mail: /users/hlein/www/webtrees-develop/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/MailTransport.php:45 The Swift_Transport_MailTransport class is deprecated since version 5.4.5 and will be removed in 6.0. Use the Sendmail or SMTP transport instead.

2)- Mail transport external (tried to use SMTP) results in
Mail: /users/hlein/www/webtrees-develop/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/StreamBuffer.php:266 Undefined variable: errno
